src/main/java/org/springframework/data/couchbase/core/mapping/BasicCouchbasePersistentEntity.java

```diff
@@ -76,24 +78,22 @@ protected CouchbasePersistentProperty returnPropertyIfBetterIdPropertyCandidateO
 		}
 
 		// check existing ID vs new candidate
-		boolean currentCbId = this.getIdProperty().isAnnotationPresent(Id.class);
 		boolean currentSpringId = this.getIdProperty().isAnnotationPresent(org.springframework.data.annotation.Id.class);
-		boolean candidateCbId = property.isAnnotationPresent(Id.class);
 		boolean candidateSpringId = property.isAnnotationPresent(org.springframework.data.annotation.Id.class);
 
-		if (currentCbId && candidateSpringId) {
-			// spring IDs will have priority over SDK IDs
+		if (candidateSpringId && !currentSpringId) {
+			// spring IDs will have priority over fields named id
 			return property;
-		} else if (currentSpringId && candidateCbId) {
-			// ignore SDK's IDs if current is a Spring ID
+		} else if (currentSpringId && !candidateSpringId) {
+			// spring IDs will have priority over fields named id
 			return null;
+		} else { // if (currentSpringId && candidateSpringId || !currentSpringId && !candidateSpringId)
```

Comment:
this comment is confusing on the else

Reply:
With two binary variables, there are four cases.   the 1 0 and 0 1 cases are above.  This else is the 0 0 and 1 1 cases.
1 1 is two fields with @Id annotation and is not allowed.  0 0 is two fields named id - which is possible through the @Field annotation.